### ISSUE TITLE
samtools view with -L and --fetch-pairs produces small --unoutput file

### ISSUE THREAD

**Issue by valeandri** [samtools/samtools#2016]
#### Are you using the latest version of [samtools](https://github.com/samtools/samtools/releases/latest) and [HTSlib]?
samtools 1.19.2
Using htslib 1.19.1

 #### Please describe your environment.
- Linux 5.15.0-100-generic
- x86_64
- gcc (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0

#### Please specify the steps taken to generate the issue, the command you are running and the relevant output.

I am trying to extract reads falling into BED regions from a cram file and save the remaining reads into a separate bam file. 
So far so good.
When I also apply the --fetch-pairs, the extracted reads are file, with the queues of the reads distribution clearly containing also the pairs, but the -U file is almost empty:
```
6,3G mar 28 11:03 NA24385_extr_negative.bam # wo --fetch-pairs
3,7M mar 28 11:50 NA24385_extr_negative.bam.bai
105K mar 29 09:03 NA24385_extr_negative_fetchpairs.bam
27K mar 29 08:54 NA24385_extr_negative_fetchpairs.bam.bai
```

The command line I used is the following:
```
samtools view NA24385/NA24385.cram -@ 4 --fetch-pairs --reference hg38_v0_Homo_sapiens_assembly38.fasta -U NA24385/NA24385_extr_negative_fetchpairs.bam -L extraction_hg38.bed -b -o NA24385/NA24385_extr_fetchpairs.bam
```
[This thread](https://github.com/samtools/samtools/pull/1542) explains the incompatibility between the --fetch-pairs and other filtering options, is this the case?

Is there a way to get the expected results?

Have a nice day,
Valentina

**Comment by whitwham** [comment]
> https://github.com/samtools/samtools/pull/1542 explains the incompatibility between the --fetch-pairs and other filtering options, is this the case?

Yes, and we document that in the man page for **samtools view** under the _--fetch-pairs_ option.

I will take a look at the code again, see if there is some way around it.

**Comment by valeandri** [comment]
Thanks!

Let me know if there is a way to cope with this limitation.

Have a nice day!

**Comment by whitwham** [comment]
I cannot see an easy way to get around this limitation.  As we document said limitation I am going to close this issue.
